### DISCUSSION TITLE
Show with maintainBufferConsistency = false and SetPixel safety

### DISCUSSION THREAD

**Discussion by awawa-dev** [Makuna/NeoPixelBus#602] [Q&A] [answered]
Hi
I have a question: is it safe to call Show with maintainBufferConsistency =false and then using SetPixel for new frame, even if canShow is still false? In the documentation only GetPixel is mentioned so can I assume it is safe and SetPixel for new colors won't affect the old frame that is still rendering at the same time? ESP32 platform.

**Comment by Makuna** [comment, marked as answer]
It can be safe.  It depends on the Method used and the hardware used.  BitBang method it will never be safe.  Most hardware supported methods (ESP32 I2s and RMT, ESP8266 DMA, etc) it will be safe.  AVR and ARM platforms it is not safe as they are just bitbang.

For the hardware methods:
If you never call `GetPixel()` then it is recommended that you pass `maintainBufferConsistency = false`.  
Internally, this option controls a buffer copy required to keep the internal double buffering consistent.  Waisted work if you never "view" the buffer.
The internal double buffering is what allows you to start updating the next "frame" while the current "frame" is being set.  Further, this double buffering is required from hardware since it uses one to actively send the data while the main sketch continues to run.

**Reply by awawa-dev** [reply]
Thank you very much for the prompt answer.
